app/Http/Controllers/Api/PostController.php

```diff
@@ -235,10 +235,37 @@ public function like(string $id): JsonResponse
             return response()->errors([], __('Post not found'), 404);
         }
 
-        $post->users()->attach(auth()->user(), ['type_interaction_id'=> 2]);
+        $user = auth()->user();
 
+        // Vérifiez si l'utilisateur a déjà aimé le post
+        $isLiked = $post->users()->where('user_id', $user->id)->exists();
 
-        return response()->success(PostResource::make($post), __('Post liked successfully'), 200);
+        DB::beginTransaction();
+
+        try {
+            if ($isLiked) {
+                // Si l'utilisateur a déjà aimé le post, retirez le like (unlike) et mettez à jour liked à false
+                $post->users()->detach($user);
+                $post->interactions()->where('user_id', $user->id)->update(['liked' => false]);
```

Comment:
upddate ([])??
Why are you changing this here. Is there an attribute liked in the table interactions?